qemu/deps/block_device/repeat_blockdev_add.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+QEMU_IMG=qemu-img
+QSD=qemu-storage-daemon
+
+if ! which $QSD;then echo "$QSD does not exist";exit 1; fi
+
+rm -f /tmp/qsd.pid
+
+"$QEMU_IMG" create -f qcow2 -F raw -b null-co:// /tmp/top.qcow2
+
+(echo '{"execute": "qmp_capabilities"}'
+ sleep 1
+ while true; do
+     echo '{"execute": "blockdev-add", "arguments": {"driver": "qcow2", "node-name": "tmp", "backing": "node0", "file": {"driver": "file", "filename": "/tmp/top.qcow2"}}}'
+     echo '{"execute": "blockdev-del", "arguments": {"node-name": "tmp"}}'
+ done) | \
+"$QSD" \
+    --chardev stdio,id=stdio \
+    --monitor mon0,chardev=stdio \
+    --object iothread,id=iothread0 \
+    --blockdev null-co,node-name=node0,read-zeroes=true \
+    --nbd-server addr.type=unix,addr.path=/tmp/nbd.sock \
+    --export nbd,id=exp0,node-name=node0,iothread=iothread0,fixed-iothread=true,writable=true \
+    --pidfile /tmp/qsd.pid \
+    &
+
+while [ ! -f /tmp/qsd.pid ]; do
+    true
+done
+
+"$QEMU_IMG" bench -f raw -c 2000000 nbd+unix:///node0\?socket=/tmp/nbd.sock
+ret=$?
+
+kill %1
+
+exit $ret
```

Comment:
Looks like a blank line needs to be added here

Reply:
updated